### ISSUE TITLE
Game crash when looking at Egyptian Artefact

### ISSUE THREAD

**Issue by dennisdahlc** [LoneGazebo/Community-Patch-DLL#2200]
Bug Report Template

Mod Version (i.e Date - (4/23b)):

7/7

1. Mod List (if using standard CPP set, leave blank):

2. Type of error (i.e. crash, interface bug, AI quirk): 

Game crash

3. Steps to reproduce:

When I mouse over the newly created artefact, the game crashes. Also when I enter the tourism window and choose "Your culture". This is the 8th burial chamber I have constructed, and nothing was wrong with the previous 7.
The game crashes every single time.

4. Additional information:

Please help me fix this. I hope I can alter my current version, since I am in a very exciting game as Egypt, and since this is gamebreaking for me since I am going for a tourism victory :)

Sry for the pdf, but I had to rename the zip, since github didn't want me to upload a zip. 

[stuff.pdf](https://github.com/LoneGazebo/Community-Patch-DLL/files/372150/stuff.pdf)

---------------------------
Supporting information:

Please note that you can attach .zip files to Issues by dragging-and-dropping them. If possible, zip up all supporting data and post that way.

1. Log files: Database.log and Lua.log needed
2. Minidump file (located in your Civ5.exe directory)
3. Screenshots (if needed)

**Comment by LoneGazebo** [comment]
You don't have logging enabled. Enable it, crash, and then re-upload.

**Comment by dennisdahlc** [comment]
Here are the logs.
I looked through them quickly and I see some errors here and there.

[Logs.pdf](https://github.com/LoneGazebo/Community-Patch-DLL/files/374171/Logs.pdf)

**Comment by LoneGazebo** [comment]
Logs look fine (most errors you see are benign, as they're vanilla things). Sorry about the CTDs. Make sure you are loading your save after loading all mods. Try loading an earlier save and see if that works.

**Comment by dennisdahlc** [comment]
Loading an older save allows me to at least continue the game, but without access to the tourism window my tourism victory is not going so well :)

Could it be something related to the name of the artefact ? When I click on the tourism icon, I can go to the Swap great works window and see all my artefacts, but I can not see my newest one which is causing the problem.

Could there be a problem in the name of the artefact perhaps ? If you need it, I can look at the current ones making it easier for you to pinpoint the exact name of the crashing artefact.

**Comment by LoneGazebo** [comment]
When you go to the window, is the UI messed up?

**Comment by dennisdahlc** [comment]
No, the GUI looks as usual except for the one artefact missing in the list.

Are the names of the artefact determined by an ordered list ? If so, I might be able to procure a save game from the turn before I actually build the burial tomb with the crashing artefact.

**Comment by LoneGazebo** [comment]
List is randomized.

**Comment by dennisdahlc** [comment]
Is there any way that I can view the list from a file ? If it is truly a specific artefact causing this, I might be able to find out which one.

**Comment by dennisdahlc** [comment]
I found a save from before the game crashing. I built a burial tomb in a city which was conquered by me, and then when I enter the tourism window, the game crashes.
I am starting to think that it is not related to a specific artefact, but perhaps when building a burial tomb in a city which was not founded by Egypt.

I hope this helps you to find the error. It would be awesome if the fixing could make me continue my current game, but since most new versions of CBO are not savegame-compatible, my hopes are not too high.

**Comment by LoneGazebo** [comment]
Shouldn't matter - how many cities do you have, may I ask?

**Comment by LoneGazebo** [comment]
May be an issue with you actually running out of great artifacts to create.

**Comment by LoneGazebo** [comment]
If so, just don't build any more burial tombs.

**Comment by dennisdahlc** [comment]
I think I have 7 that I founded myself and build burial tombs in them. I probably conquered 5 which already had burial tombs but with no artefacts, since I didn't build them myself. The crashing city might be the 12th or so city.
Too bad if I have to stop building burial tombs, but I guess that is better than not being able to play the game :)

**Comment by LoneGazebo** [comment]
How did other civ cities have burial tombs in them?

**Comment by pandasnail** [comment]
Caravansaries becoming Tombs on conquest is I assume what he means

**Comment by dennisdahlc** [comment]
Pandasnail is correct. The crashing city would be the 8th city I have build a burial tomb in, although I have not settled it myself. Perhaps that is the problem.

If I settle another city and builds a burial tomb in that, would it then also crash ? I will report if I find out.

**Comment by HalfEmptyMug** [comment]
I'm having the same problem, it doesn't seem to matter whether the offending burial tomb is built in a city you've settled or conquered.

**Comment by dennisdahlc** [comment]
Did you manage to fix it ?

**Comment by LoneGazebo** [comment]
We shall see. I added some checks to keep the game from making a bad artifact.

**Comment by HalfEmptyMug** [comment]
I'm not sure if this is the best place to write this since this is now closed, but after some testing with IGE I believe I found the culprit of the crash. It seems that after _any_ civilization discovers archaeology all artifacts in the world are set and no more can be created. When building a burial tomb before this discovery an artifact appears as normal, but after it is discovered no artifact appears (which would result in a crash in the older version). 
Also, in the latest version (7-27) a burial tomb cannot be constructed if the game starts in the industrial era or later, but other civs may build caravansaries (which, similar to religion being turned off, would explain why a building creating an artifact would be unavailable).

The checks you added seemed to resolve the issue just fine, but this might shed light on any other issues that may arise regarding artifact creation.
